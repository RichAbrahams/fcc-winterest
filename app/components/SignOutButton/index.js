/**
*
* SignOutButton
*
*/

import React from 'react';
import FA from './styles';


function SignOutButton({ children, signOut }) {
  return (
    <FA
      onClick={() => signOut()}
      name="sign-out"
      size="3x"
    >
      { children }
    </FA>
  );
}

SignOutButton.propTypes = {
  children: React.PropTypes.array,
  signOut: React.PropTypes.function,
};

export default SignOutButton;
