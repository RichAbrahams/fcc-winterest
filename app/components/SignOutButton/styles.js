import styled from 'styled-components';
import FontAwesome from 'react-fontawesome';

const FA = styled(FontAwesome)`
  color: white;
  margin: 0rem 3rem 0rem 2rem;
`;

export default FA;

