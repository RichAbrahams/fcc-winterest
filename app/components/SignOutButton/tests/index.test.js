import React from 'react';
import renderer from 'react-test-renderer';
import SignOutButton from '../index';

it('renders correctly', () => {
  const tree = renderer.create(
    <SignOutButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
