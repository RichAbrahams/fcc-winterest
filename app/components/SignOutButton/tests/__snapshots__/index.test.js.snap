exports[`test renders correctly 1`] = `
<span
  aria-hidden={true}
  className="fa fa-sign-out fa-3x frqUSI"
  onClick={[Function]} />
`;
